### DISCUSSION TITLE
Align progress bar in console?

### DISCUSSION THREAD

**Discussion by zachlim98** [Textualize/rich#1598] [General]
I've played around with using `Progress` and `BarColumn` to customize tracking but I can't seem to align the progress bars. I tried to use `Align.center` as below but got an error when rendering - anyone has any ideas? Thanks! :)

```python
text_column = TextColumn("{task.description}", table_column=Column(ratio=1))
bar_column = BarColumn(bar_width=None, table_column=Column(ratio=2))
spinner_column = SpinnerColumn()
progress = Align.center(Progress(text_column, bar_column, spinner_column, expand=False))

with progress:
    for n in progress.track(range(100)):
        sleep(0.1)
```

**Comment by willmcgugan** [comment]
You could extend the BarColumn and add the Align to the `render()` method.

Something like this (untested):

```python
class CenteredBarColumn(BarColumn):
    def render(self, task: "Task") -> RenderableType:
        return Align.center(super().render(task))
```

**Reply by willmcgugan** [reply]
Unless you wanted to align the entire thing? In that case you can extend the Progress class itself.

```python
class CenteredProgress(Progress):
    def get_renderable(self) -> RenderableType:
        return Align.center(super().get_renderable())
```

**Reply by zachlim98** [reply]
Yes - centering the whole thing would be it. Thanks for the reply, will try that out and let you know how it goes
